### PR TITLE
Add is_active flag for registration codes.

### PR DESCRIPTION
**Pull request by mlen108**

### CODE REVIEW

**Comment by mlen108** [comment]
@maxpeterson ^^

File: profiles/modules/codes/fields.py [hunk 1]
```diff
@@ -10,10 +10,13 @@ class CodeField(fields.CharField):
...
 
     def to_python(self, value):
+        if not value:
+            return value
+
         try:
             key = 'code__iexact'
             value = self.re_clean_code.sub('', value)
-            value = Code.objects.get(**{key: value})
+            value = Code.objects.get(**{key: value, 'is_active': True})
```
Comment (maxpeterson) [inline profiles/modules/codes/fields.py]: This seems over complicated. Why is it not `value = Code.objects.get(code__iexact=value, is_active= True)`?

File: profiles/modules/codes/fields.py [hunk 2]
```diff
@@ -10,10 +10,13 @@ class CodeField(fields.CharField):
     re_clean_code = re.compile('[^a-zA-Z0-9]+')
 
     def to_python(self, value):
+        if not value:
+            return value
+
         try:
             key = 'code__iexact'
             value = self.re_clean_code.sub('', value)
```
Comment (maxpeterson) [inline profiles/modules/codes/fields.py]: Why is it removing non word characters? I am not sure it is a bad thing, just wandered if there was a specific reason.

**Comment by maxpeterson** [comment]
I can't remember if this was ready to merge or not, but it looks fine to me.

File: profiles/modules/codes/fields.py (hunk 2 above)
Comment (mlen108) [inline profiles/modules/codes/fields.py]: This regex comes from an old code... So then should it remove all whitespaces or just strip() ?

File: profiles/modules/codes/fields.py (hunk 1 above)
Comment (mlen108) [inline profiles/modules/codes/fields.py]: Fixed in 0b858f9

File: profiles/modules/codes/fields.py [hunk 3]
```diff
@@ -10,10 +10,12 @@ class CodeField(fields.CharField):
     re_clean_code = re.compile('[^a-zA-Z0-9]+')
```
Comment (maxpeterson) [inline profiles/modules/codes/fields.py]: Change to only remove whitespace.

**Comment by mlen108** [comment]
It's ready to merge now.

**Pull request merged**